[ET-VK][Ops] dequantize ops skeleton test framework

ahmtox opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11553
* #11557
* #11483
* #11369
* #11556
* #11555
* #11554
* #11482
* #11481
* __->__ #11480
* #11368
* #11367
* #11366
* #11575
* #11365
* #11552
* #11479

Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

Commit:
[ET-VK][Ops] dequantize ops skeleton test framework

Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

manuelcandales commented on backends/vulkan/test/op_tests/dequantize_test.cpp:
```diff
@@ -0,0 +1,316 @@
+//
+
+// Helper function to get the name of a ScalarType for better error messages
+std::string scalar_type_name(c10::ScalarType dtype) {
```
isn't this the same helper function you used for the quantized tests? if so, please refactor this into some utilities.

manuelcandales commented on backends/vulkan/test/op_tests/dequantize_test.cpp:
```diff
@@ -0,0 +1,316 @@
+  }
+}
+
+vkcompute::vkapi::ScalarType from_at_scalartype(c10::ScalarType at_scalartype) {
```
same as above, refactor into common utilities

manuelcandales commented on backends/vulkan/test/op_tests/dequantize_test.cpp:
```diff
@@ -0,0 +1,316 @@
+    at::ScalarType dtype,
+    at::ScalarType out_dtype) {
+  auto out = at::empty_like(input, out_dtype);
+  // Convert at::ScalarType to executorch::ScalarType
```
you use this conversion here two times, and also in the quantized tests, please refactor into a utility function

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize ops skeleton test framework"


Skeleton framework that is needed to build out the dequantize_per_tensor and dequantize_per_token operators based on cpu implementation

Differential Revision: [D76267021](https://our.internmc.facebook.com/intern/diff/D76267021/)

[ghstack-poisoned]
